--- media/db/mysql_kalkun.sql	2020-01-31 10:31:25.170210863 +0100
+++ media/db/mysql_kalkun.sql	2020-01-31 10:49:28.860029905 +0100
@@ -182,6 +182,20 @@
 
 -- --------------------------------------------------------
 
+CREATE TABLE IF NOT EXISTS `pbk` (
+  `ID` integer NOT NULL auto_increment,
+  `GroupID` integer NOT NULL default '-1',
+  `Name` text NOT NULL,
+  `Number` text NOT NULL,
+  PRIMARY KEY (`ID`)
+);
+
+CREATE TABLE IF NOT EXISTS `pbk_groups` (
+  `Name` text NOT NULL,
+  `ID` integer NOT NULL auto_increment,
+  PRIMARY KEY `ID` (`ID`)
+);
+
 --
 -- Alter table structure for table `pbk`
 --
@@ -255,9 +269,9 @@
 CREATE TABLE IF NOT EXISTS `plugins` (
   `plugin_id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
   `plugin_system_name` varchar(255) NOT NULL,
-  `plugin_name` varchar(255) NOT NULL,
+  `plugin_name` varchar(255) NULL,
   `plugin_uri` varchar(120) DEFAULT NULL,
-  `plugin_version` varchar(30) NOT NULL,
+  `plugin_version` varchar(30) NULL,
   `plugin_description` text,
   `plugin_author` varchar(120) DEFAULT NULL,
   `plugin_author_uri` varchar(120) DEFAULT NULL,
