--- /dev/null	2024-11-05 10:18:18.264113631 +0000
+++ application/models/Api_model.php	2024-11-05 15:09:39.317167125 +0000
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Api_model Class
+ *
+ * Define methods for plugins
+ *
+ * @package	Kalkun
+ * @subpackage	Api
+ * @category	Models
+ */
+class Api_model extends CI_Model {
+
+	/**
+	 * Constructor
+	 *
+	 * @access	public
+	 */
+	function __construct()
+	{
+		parent::__construct();
+	}
+
+	// --------------------------------------------------------------------
+
+	/**
+	 * Login
+	 *
+	 * Check login credential
+	 *
+	 * @access	public
+	 */
+	function login($credentials = false)
+	{
+		if ($credentials === false) return false;
+
+		$username = (isset($credentials['user'])) ? $credentials['user'] : false;
+		$password = (isset($credentials['pass'])) ? $credentials['pass'] : false;
+
+		if (empty($username) || $username === false || empty($password) || $password === false) return false;
+
+		$this->db->from('user');
+		$this->db->where('username', $username);
+		$query = $this->db->get();
+
+		if($query->num_rows()=='1' && password_verify($password, $query->row('password'))) {
+			return true;
+		}
+		else return false;
+	}
+
+}
+
+/* End of file api_model.php */
+/* Location: ./application/models/api_model.php */
