--- system/libraries/Loader.php	2020-01-31 11:26:58.000000000 +0100
+++ system/libraries/Loader.php	2020-01-31 13:42:11.795159364 +0100
@@ -830,6 +830,28 @@
 			}
 		} // END FOREACH
 
+		// Try to check with kalkun plugins logic
+		$backtrace = debug_backtrace();
+		foreach ($backtrace as $k => $b) {
+			if ($k >= 2) break;
+			if (isset($b['file']) && preg_match(';/application/plugins/;', $b['file'])) {
+				$filepath = array();
+				foreach (array($class, ucfirst($class), strtolower($class)) as $class) {
+					$filepath[] = dirname($b['file']).'/libraries/'.$class.EXT;
+					$filepath[] = dirname($b['file']).'/../libraries/'.$class.EXT;
+					$filepath[] = dirname($b['file']).'/'.$class.EXT;
+					$filepath[] = dirname($b['file']).'/../'.$class.EXT;
+					foreach ($filepath as $f) {
+						//echo $f."<br>";
+						if ( file_exists($f) ) {
+							$backtrace = NULL;
+							return $this->_ci_init_class($class, dirname($f).'/', $params, $object_name);
+						}
+					}
+				}
+			}
+		}
+		
 		// One last attempt.  Maybe the library is in a subdirectory, but it wasn't specified?
 		if ($subdir == '')
 		{
--- /dev/null	2020-01-31 15:28:02.904385989 +0100
+++ application/models/api_model.php	2020-01-31 14:46:49.430161635 +0100
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Kalkun
+ * An open source web based SMS Management
+ *
+ * @package		Kalkun
+ * @author		Kalkun Dev Team
+ * @license		http://kalkun.sourceforge.net/license.php
+ * @link		http://kalkun.sourceforge.net
+ */
+
+// ------------------------------------------------------------------------
+
+/**
+ * Api_model Class
+ *
+ * Define methods for plugins 
+ *
+ * @package	Kalkun
+ * @subpackage	Api
+ * @category	Models
+ */
+class Api_model extends Model {
+
+	/**
+	 * Constructor
+	 *
+	 * @access	public
+	 */	
+	function Api_model()
+	{
+		parent::Model();
+	}
+	
+	// --------------------------------------------------------------------
+	
+	/**
+	 * Login
+	 *
+	 * Check login credential
+	 *
+	 * @access	public   		 
+	 */	
+	function login($credentials = false)
+	{
+		if ($credentials === false) return false;
+		
+		$username = (isset($credentials['user'])) ? $credentials['user'] : false;
+		$password = (isset($credentials['pass'])) ? sha1($credentials['pass']) : false;
+		
+		if (empty($username) || $username === false || empty($password) || $password === false) return false;
+		
+		$this->db->from('user');
+		$this->db->where('username', $username);
+		$this->db->where('password', $password);
+		$query = $this->db->get();
+		
+		if($query->num_rows()=='1') {
+			return true;
+		}
+		else return false;
+	}
+	
+}
+
+/* End of file api_model.php */
+/* Location: ./application/models/api_model.php */
