--- application/plugins/jsonrpc/controllers/Jsonrpc.php	2024-11-05 12:43:31.658458928 +0000
+++ application/plugins/jsonrpc/controllers/Jsonrpc.php	2024-11-05 15:48:18.618983109 +0000
@@ -26,6 +26,35 @@
 	function __construct()
 	{
 		parent::__construct(FALSE);
+
+		// Add login check to this plugin
+		$this->load->model('Api_model');
+		$data = @json_decode(file_get_contents("php://input"), true);
+		$id = (isset($data['id']) && isset($data['id'])) ? $data['id'] : NULL;
+		$credentials['user'] = (isset($data['params']) && isset($data['params']['user'])) ? $data['params']['user'] : false;
+		$credentials['pass'] = (isset($data['params']) && isset($data['params']['pass'])) ? $data['params']['pass'] : false;
+		if (! $this->Api_model->login($credentials)) {
+			$content = json_encode(array(
+				'jsonrpc' => Datto\JsonRpc\Server::VERSION,
+				'id' => $id,
+				'error' => array(
+					'code'=> 000,
+					'message'=> 'Please provide valid login credentials!'
+				)
+			));
+
+			header('HTTP/1.1 401 Unauthorized');
+
+			if ($id === NULL) {
+				# Without id, no additional response is wanted by user
+				die();
+			}
+			else {
+				header('Content-Type: application/json');
+				header('Content-Length: ' . strlen($content));
+				die($content);
+			}
+		}
 	}

 	/**
